### PR TITLE
internal/repos: Better error messaging for missing SourceInfo

### PR DESCRIPTION
**Pull request by indradhanush**
A [customer](https://github.com/sourcegraph/customer/issues/1688) is affected by this error.  Better messaging might have helped us understand it better.

## Test plan

No functional change.

### CODE REVIEW

File: internal/repos/store.go [hunk 1]
```diff
@@ -370,8 +370,10 @@ func (s *store) CreateExternalServiceRepo(ctx context.Context, svc *types.Extern
 	)
 
 	src := r.Sources[svc.URN()]
-	if src == nil || src.CloneURL == "" {
-		return errors.New("CreateExternalServiceRepo: repo missing source info for external service")
+	if src == nil {
+		return errors.New("CreateExternalServiceRepo: nil repo for external service")
```
Comment (sashaostrikov) [inline internal/repos/store.go:374]: I kind of liked "repo missing source info for external service" for this case more, but that's probably just me :)

**Review by sashaostrikov — APPROVED** [review]

File: internal/repos/store.go (hunk 1 above)
Comment (indradhanush) [reply internal/repos/store.go:374]: I'll revert  that bit.

File: internal/repos/store.go [hunk 2]
```diff
@@ -370,8 +370,10 @@ func (s *store) CreateExternalServiceRepo(ctx context.Context, svc *types.Extern
 	)
 
 	src := r.Sources[svc.URN()]
-	if src == nil || src.CloneURL == "" {
+	if src == nil {
 		return errors.New("CreateExternalServiceRepo: repo missing source info for external service")
+	} else if src.CloneURL == "" {
```
Comment (ryanslade) [inline internal/repos/store.go:375]: Maybe we should include the repo id in both messages?

Comment (indradhanush) [reply internal/repos/store.go:375]: Good idea. PR coming in 3, 2, 1...

Comment (indradhanush) [reply internal/repos/store.go:375]: https://github.com/sourcegraph/sourcegraph/pull/45709

File: internal/repos/store.go [hunk 3]
```diff
@@ -370,8 +370,10 @@ func (s *store) CreateExternalServiceRepo(ctx context.Context, svc *types.Extern
...
+		return errors.Newf("CreateExternalServiceRepo: repo (ID=%q) missing CloneURL for external service", src.ID)
```
Comment (mrnugget) [inline internal/repos/store.go:376]: `src.ID` is an `int64`, why quote it?

Comment (indradhanush) [reply internal/repos/store.go:376]: This is getting fixed in #45709.

**Pull request merged**